buildSrc/subprojects/plugins/src/main/kotlin/org/gradle/gradlebuild/unittestandcompile/UnitTestAndCompilePlugin.kt

```diff
@@ -45,15 +45,15 @@ import testLibrary
 import java.util.jar.Attributes
 
 
-enum class ModuleType(val source: JavaVersion, val target: JavaVersion, val requiredCompiler: JavaVersion) {
-    UNDEFINED(JavaVersion.VERSION_1_1, JavaVersion.VERSION_1_1, JavaVersion.VERSION_1_1),
-    ENTRY_POINT(JavaVersion.VERSION_1_6, JavaVersion.VERSION_1_6, JavaVersion.VERSION_1_6),
-    WORKER(JavaVersion.VERSION_1_6, JavaVersion.VERSION_1_6, JavaVersion.VERSION_1_6),
-    CORE(JavaVersion.VERSION_1_7, JavaVersion.VERSION_1_7, JavaVersion.VERSION_1_7),
-    PLUGIN(JavaVersion.VERSION_1_7, JavaVersion.VERSION_1_7, JavaVersion.VERSION_1_7),
-    INTERNAL(JavaVersion.VERSION_1_7, JavaVersion.VERSION_1_7, JavaVersion.VERSION_1_7),
-    REQUIRES_JAVA_8(JavaVersion.VERSION_1_8, JavaVersion.VERSION_1_8, JavaVersion.VERSION_1_8),
-    REQUIRES_JAVA_9_COMPILER(JavaVersion.VERSION_1_6, JavaVersion.VERSION_1_6, JavaVersion.VERSION_1_9);
+enum class ModuleType(val source: JavaVersion, val target: JavaVersion) {
+    UNDEFINED(JavaVersion.VERSION_1_1, JavaVersion.VERSION_1_1),
+    ENTRY_POINT(JavaVersion.VERSION_1_6, JavaVersion.VERSION_1_6),
+    WORKER(JavaVersion.VERSION_1_6, JavaVersion.VERSION_1_6),
+    CORE(JavaVersion.VERSION_1_7, JavaVersion.VERSION_1_7),
+    PLUGIN(JavaVersion.VERSION_1_7, JavaVersion.VERSION_1_7),
+    INTERNAL(JavaVersion.VERSION_1_7, JavaVersion.VERSION_1_7),
+    REQUIRES_JAVA_8(JavaVersion.VERSION_1_8, JavaVersion.VERSION_1_8),
+    REQUIRES_JAVA_9_COMPILER(JavaVersion.VERSION_1_6, JavaVersion.VERSION_1_6);
```

Comment:
If we now always require a Java 9 compiler, then I guess we can remove that enum constant? It will probably be an entry point, right?

Reply:
Unfortunately we can't right now, at lease some Java 7 subprojects are using `<>` operator, which requires `sourceCompatiblity=1.7`. Let's not touch this part for now.